google/cloud/aiplatform/models.py

```diff
@@ -704,6 +750,13 @@ def _deploy_call(
                 display_name=deployed_model_display_name,
             )
 
+        # Service will throw error if both metadata and parameters are not provided
+        if explanation_metadata or explanation_parameters:
+            explanation_spec = gca_endpoint.explanation.ExplanationSpec()
+            explanation_spec.metadata = explanation_metadata
+            explanation_spec.parameters = explanation_parameters
+            deployed_model.explanation_spec = explanation_spec
+
```

Comment:
The thought behind this was allowing the service to throw the exception about them being passed together instead of creating two sets of validation. I confirmed that an error response is returned service-side right away if only one is passed.

Reply:
We inform the user in the docstring. Do you think we should add validation in the library as well?
https://github.com/googleapis/python-aiplatform/blob/b5000c2d0f27c4a746bb07c499918d54092a7458/google/cloud/aiplatform/models.py#L480-L481

Reply:
I think telling users what will fail as early as possible will be better. We don't necessary need full validation (though it is good to have if it is low cost), but having some basic sanity check can help prevent user frustration (especially since all the operations here are async and have some wait time). 

Also, I noticed that there other similar logics are using `and` (e.g., Ln 1287), so I thought it was unintentional :)

Reply:
Good point, I've added the local validation for ensuring they are passed together.